sunpy/timeseries/timeseriesbase.py

```diff
@@ -524,6 +525,35 @@ def _sanitize_metadata(self, **kwargs):
         redundant_cols = list(set(self.meta.columns) - set(self.columns))
         self.meta._remove_columns(redundant_cols)
 
+    def power_spectra(self, fs = 1):
+        """
+        Estimate power spectral density of the TimeSeries
+
+        Parameter
+        ----------
+        fs : `float, optional`
+            Sampling frequency of the time series in units of Hz.
+            Defaults to 1.0.
+
+        Returns
+        -------
+        Freqs : array of arrays
+            Array of sample frequencies for all the columns in the TimeSeries.
+
+        Spectra : array of arrays
+            Power spectral density or power spectrum of all the columns in
+            the TimeSeries.
+        """
+        Freqs = []
+        Spectra = []
+        for col in self.columns:
+            data_ = self.data[col]
+            freq, spectra = signal.periodogram(data_, fs)
+            Freqs.append(freq)
+            Spectra.append(spectra)
```

Comment:
```python
from functools import partial
periodogram_fs = partial(signal.periodogram, fs=fs)
Freqs, Spectra = zip(*map(periodogram_fs, self.data))
```
This will remove all the loop, it will produce tuples instead of lists though.